### ISSUE TITLE
[Windows] Failed to run ESM, throwing ERR_UNSUPPORTED_ESM_URL_SCHEME

### ISSUE THREAD

**Issue by compulim** [fgnass/node-dev#266]
## Description

When running `node-dev ./index.mjs` on Windows, Node.js (14.* and 16.*) will throw `ERR_UNSUPPORTED_ESM_URL_SCHEME`. It works on Ubuntu.

[According to doc](https://github.com/fgnass/node-dev#esmodules), I renamed to `.mjs` and I believe this is the only change needed for ESM.

> When using ESModule syntax and `.mjs` files, `node-dev` will automatically use a loader to know which files to watch.

## Repro steps

I created a repo for bug repro, with GitHub Actions configured to run on both [Ubuntu](https://github.com/compulim/experiment-node-dev-esm/actions/workflows/run-ubuntu.yml) and [Windows](https://github.com/compulim/experiment-node-dev-esm/actions/workflows/run-windows.yml).

```
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'd:'
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:791:11)
    at Loader.resolve (internal/modules/esm/loader.js:89:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
    at Loader.import (internal/modules/esm/loader.js:177:28)
    at internal/process/esm_loader.js:57:31
    at initializeLoader (internal/process/esm_loader.js:62:5)
    at Object.loadESM (internal/process/esm_loader.js:67:11)
    at runMainESM (internal/modules/run_main.js:47:31)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:73:5)
    at internal/main/run_main_module.js:17:47 {
  code: 'ERR_UNSUPPORTED_ESM_URL_SCHEME'
}
```

The workflow runs the following steps:

1. Install Node.js (12.x, 14.x, 16.x)
1. `npm install` (with `node-dev@^7.0.0`)
1. `npm start`
   - It runs `node-dev ./index.mjs`
1. When `index.mjs` is loaded, it `import hello from './hello.mjs'`

To repro without a Windows box, please fork the repo and see the result of GitHub Actions. 😉

Instead of `npm start`, I also tried the followings and got same error:

- `node_modules\.bin\node-dev .`
- `node_modules\.bin\node-dev ./index.mjs`

**Comment by bjornstar** [comment]
According to https://github.com/nodejs/node/issues/31710, we should use `pathToFileUrl`.

It's probably a good idea to add a windows build in our github actions.

**Comment by bjornstar** [comment]
I've added windows to the tests and can see the ESM tests are failing. I'll take a look at it when I get some more time.

**Comment by kherock** [comment]
I believe this is resolved by #279. Using `fileURLToPath` in the ESM loader seems to have fixed those tests on Windows

**Comment by bjornstar** [comment]
Tests are looking great, thanks @kherock. I'll go ahead and cut a release. Let's call this fixed in v7.4.0
